src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -14,52 +14,72 @@ public class TimelineViewModel : ReactiveObject
     {
         private DateTime _lastDateLoaded;
 
-        private List<Toggl.TimelineChunkView> _timelineChunks;
-        private List<Toggl.TogglTimeEntryView> _timeEntries;
-
         public TimelineViewModel()
         {
             this.WhenAnyValue(x => x.RecordActivity).ObserveOn(RxApp.TaskpoolScheduler)
                 .Subscribe(value => Toggl.SetTimelineRecordingEnabled(value));
             OpenTogglHelpUri = ReactiveCommand.Create(() =>
                 Toggl.OpenInBrowser("https://support.toggl.com/en/articles/3836325-toggl-desktop-for-windows"));
-            this.WhenAnyValue(x => x.SelectedDate).Subscribe(HandleSelectedDateChanged);
+
+            Toggl.TimelineSelectedDate.Subscribe(date => SelectedDate = date);
+
+            this.WhenAnyValue(x => x.SelectedDate)
+                .Where(date => date != Toggl.TimelineSelectedDate.Value)
+                .ObserveOn(RxApp.TaskpoolScheduler)
+                .Subscribe(date => Toggl.SetViewTimelineDay(Toggl.UnixFromDateTime(date)));

```

Comment:
Why do we need this condition `.Where(date => date != Toggl.TimelineSelectedDate.Value)`? Anyway we synchronize these 2 dates one line above. If date `Toggl.TimelineSelectedDate` would emit next value because of TE change, not because of date change, then `SelectedDate` would just stay the same and nothing would happen.

Reply:
Because in case `Toggl.TimelineSelectedDate` emits a value, different from `SelectedDate`, we would do an unnecessary call of `Toggl.SetViewTimelineDay()`. I don't know if such a situation is possible, but I think it's bad to rely on the fact that it's impossible as it's the logic external to the `TimelineViewModel`. The `Where` clause is fairly simple and fast anyway.